src/main/java/io/netty/incubator/codec/http3/DefaultHttp3Headers.java

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright 2020 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.incubator.codec.http3;
+
+import io.netty.handler.codec.CharSequenceValueConverter;
+import io.netty.handler.codec.DefaultHeaders;
+import io.netty.util.AsciiString;
+import io.netty.util.ByteProcessor;
+import io.netty.util.internal.PlatformDependent;
+import io.netty.util.internal.UnstableApi;
+
+import static io.netty.incubator.codec.http3.Http3Headers.PseudoHeaderName.hasPseudoHeaderFormat;
+import static io.netty.util.AsciiString.CASE_INSENSITIVE_HASHER;
+import static io.netty.util.AsciiString.CASE_SENSITIVE_HASHER;
+import static io.netty.util.AsciiString.isUpperCase;
+
+@UnstableApi
+public class DefaultHttp3Headers
+        extends DefaultHeaders<CharSequence, CharSequence, Http3Headers> implements Http3Headers {
+    private static final ByteProcessor HTTP3_NAME_VALIDATOR_PROCESSOR = new ByteProcessor() {
+        @Override
+        public boolean process(byte value) {
+            return !isUpperCase(value);
+        }
+    };
+    static final NameValidator<CharSequence> HTTP3_NAME_VALIDATOR = new NameValidator<CharSequence>() {
+        @Override
+        public void validateName(CharSequence name) {
+            if (name == null || name.length() == 0) {
+                PlatformDependent.throwException(
+                    new Http3Exception(String.format("empty headers are not allowed [%s]", name)));
```

Comment:
let's update the `validateName` methods signature to include `throws` declaration so we don't need to use this hack

Reply:
Errors/exceptions handling definitely needs so love here

Reply:
This one turned out to be a pretty deep hole: it relies on an interface from `codec`. I'm going to just leave it "as is" for now. We can change it later